nemo/collections/common/tokenizers/tokenizer_spec.py

```diff
@@ -49,3 +49,9 @@ def ids_to_text(self, ids):
 
     def add_special_tokens(self, special_tokens: List[str]):
         raise NotImplementedError("To be implemented")
+
+    def tokens_to_ids(self, tokens: Union[str, List[str]]) -> Union[int, List[int]]:
+        ids = []
+        for token in tokens:
+            ids.append(self.token_to_id(token))
+        return ids
```

Comment:
please see the previous implementation in the Sentence piece - tokens could be either str or List - I'd leave this change for a separate PR

Reply:
yes not for this PR

Reply:
then please revert the removal from SPE

Reply:
why? we are not using it anywhere. we have both the function tokens_to_ids and token_to_id.
Its really bad software design to let token_to_id take a list instead of a str, if you want a list use tokens_to_ids